inc/classes/subscriber/preload/class-preload-subscriber.php

```diff
@@ -40,10 +51,12 @@ class Preload_Subscriber implements Subscriber_Interface {
 	 * @author Remy Perona
 	 *
 	 * @param Homepage     $homepage_preloader Homepage Preload instance.
+	 * @param Sitemap      $sitemap_preloader  Sitemap Preload instance.
 	 * @param Options_Data $options            WP Rocket Options instance.
 	 */
-	public function __construct( Homepage $homepage_preloader, Options_Data $options ) {
+	public function __construct( Homepage $homepage_preloader, Sitemap $sitemap_preloader, Options_Data $options ) {
 		$this->homepage_preloader = $homepage_preloader;
+		$this->sitemap_preloader  = $sitemap_preloader;
```

Comment:
Let's talk about this solution:

1. We are creating a new instance of the `Sitemap` when we inject it into the homepage's subscriber.
2. We are mixing concerns as these are different.
3. And we're only doing it because we need to add them together for the counter.

I propose that we look at create a different approach to handle the Preload Count notice separately. Let it pull from homepage, sitemap, or both.

What do you think @Screenfeed @Tabrisrp ?

Reply:
From our Slack discussion, we agreed to simplify this:

1. Don't inject the Sitemap Preloader.
2. Combine the preload transients into one shared preload count transient.
3. Get the transient directly using `get_transient()` to refresh the admin notice.

Reply:
So after more discussion, we keep the two transients because they are needed as explained in #2342: to prevent sitemap preload to reset homepage’s counter, and because homepage and sitemap preload are actually working asynchronously.